### PR TITLE
Updates from RRFS_A v0.7.3 retro tests 

### PR DESCRIPTION
**Pull request by hu5970**
The current workflow is tested on Dogwood with retrospective experiments and compared with the GSL RRFS_dev1 tag 0.7.1.
Several small updates needed to match the current RRFS_A v0.7.1:
1) turn on OPM thread by updating model_configure and nems configure.
2) set tg3_from_soil=True for make_ics
3) update configure files for RRFS_A retro runs.
4) changes in prep_cyc for initial Jet/Hera runs.

This PR addresses issue #92 

The retro experiments are conducted on Dogwood and got identical results between v0.7.1 and v0.7.3 for control cycle and ensemble DA cycle in initial/boundary, GSI/EnKF analysis, nonvar cloud analysis, data process, and model forecast restart files.

Also, the wall time for each components are similar to the v0.7.1.

### CODE REVIEW

**Comment by hu5970** [comment]
@chan-hoo @MatthewPyle-NOAA Please review this PR. Hopefully, we can build v0.7.3 based on this PR.

File: parm/nems.configure [hunk 1]
```diff
@@ -2,14 +2,6 @@
 ####  NEMS Run-time Configuration File  #####
 #############################################
 
-# ESMF #
```
Comment (chan-hoo) [inline parm/nems.configure:5]: @hu5970, do you have any reason for removing this part. This if-statement can keep the ufs-weather-model printing out a large number of PET* log files.

**Review by MatthewPyle-NOAA — APPROVED** [review]
No red flags for me based on a quick look.  But would like to see approval from someone else as well before merging.

File: parm/nems.configure (hunk 1 above)
Comment (hu5970) [reply parm/nems.configure:5]: Yes, this part somehow blocked OMP to use multiple threads. I do not know why but we have to live with a large number of PET* log files for now.

Comment (MatthewPyle-NOAA) [reply parm/nems.configure:5]: @chan-hoo Are you okay with it despite this little annoyance?

**Pull request merged**